app/api/v2/queries/user_with_label_filter.rb

```diff
@@ -0,0 +1,44 @@
+# queries helping module
+module API::V2::Queries
+  class UserWithLabelFilter
+    attr_accessor :initial_scope
+
+    # initialize query to get User.all
+    def initialize(initial_scope)
+      @initial_scope = initial_scope.left_outer_joins(:labels)
+    end
+
+    # returns query with with all applied filters
+    def call(params)
+      scoped = filter_by_date(initial_scope, params[:range], params[:from], params[:to])
+      scoped = filter_by_key(scoped, params[:key])
```

Comment:
a bit shorter https://gist.github.com/denisfd/01cb2da9f04a9119da1c721df4fa8de8

Reply:
@denisfd for sure, it is. But I prefer to keep this per function filtering.
As soon as we still build ONE query it is more about readability.

Your suggestion is OK, but it not what we likely want to have in every query filter class. If you check others (that already in master) you will see we can have up to 10 filter functions. And in this case your approach starts to work as "magic", which is not what we want to have in open source project